### ISSUE TITLE
Add Timestamp Editing by hooking: sceRtcGetCurrentNetworkTick 

### ISSUE THREAD

**Issue by SilicaAndPina** [SilicaAndPina/TropHAX#22]
Title.

**Comment by 4rchitect** [comment]
How do you plan to do it?

It would work if overwrite sceRtcGetCurrentNetworkTick to return the date we want?

**Comment by SilicaAndPina** [comment]
Overwrite the function? Yea don't think so C is not JavaScript you need to hook the function to change the stamp (ret value) its formatted like every other PS Timestamp .

**Comment by 4rchitect** [comment]
Are you asking someone to work on it? If so I would like to try! :)

**Comment by SilicaAndPina** [comment]
mhm i was kinda thinking of trying myself. but you can if you want xD

**Comment by SilicaAndPina** [comment]
also the nids are on db.yml.
          sceRtcGetCurrentNetworkTick: 0xCDDD25FE

this returns a PS Timestamp, setTick sets the normal clock (i think?) and even if it didnt it would be a pretty poor solution anyway, why should the users system time be messed up just to unlock trophy's?
sceRtcAddSeconds adds x seconds to an EXISTING timestamp. not to the actural system-wide ticks.

**Comment by 4rchitect** [comment]
seems I Can't even hook the function :(

The code I wrote to hook it looks like this:
`hook[10] = taiHookFunctionImport( &ref_hook5, TAI_MAIN_MODULE, TAI_ANY_LIBRARY, 0xCDDD25FE, sceRtcGetCurrentNetworkTick_patched);`

Am I doing it right?

**Comment by SilicaAndPina** [comment]
^ It doesn't return ptick.
and your doing it from TAI_MAIN_MODULE which wont work.

**Comment by Her0es** [comment]
Good work on the plugin @SilicaAndPina 
If you do make progress on the time stamp thing, i’ll Be more than happy to test it. From what I’ve seen it’s been a long sought after feature.

**Comment by 4rchitect** [comment]
As far as I understood from the docs, sceRtcGetCurrentNetworkTick pass the value of tick by reference. And what sceRtcGetCurrentNetworkTick returns is an int to check if everything is ok.
The docs I read to get the information are here: https://docs.vitasdk.org/group__SceRtcUser.html
SceRtcTick is a struct that has a variable tick inside it and the variable tick is  unsigned long long type, so I'm pretty sure sceRtcGetCurrentNetworkTick doesn't return the actual tick because it returns an int.

I'm falling on put in the right module. I checked other codes and all of them use TAI_MAIN_MODULE.
I just found this git using a different approach to loading a module. I'll try it later the link is below:
https://github.com/theorywrong/LastSeconds/blob/master/main.c
the line is 188.

I also wrote some helper functions to translate SceDateTime to tick and I would like to know if I can share my work with you.

**Comment by SilicaAndPina** [comment]
yes but you wrote return sceRtcGetCurrentNetworkTick(PTick); which doesnt make much sense

**Comment by SilicaAndPina** [comment]
you can share your helper functions tho theres allready a function on vita to do that..

**Comment by 4rchitect** [comment]
Sorry for my bad mistakes @SilicaAndPina, and thanks for give me hints!

https://pastebin.com/KrHU65xe

Still no progress. Maybe this module is not loaded at application startup? If so, I suppose to use the Hooking weak imports method (https://tai.henkaku.xyz/docs/md__u_s_a_g_e.html). But how to figure out the module ID?

Thanks for your time and patience!

**Comment by SilicaAndPina** [comment]
Blame Rinnie, SceRtcGetCurrentNetworkTick is acturally in SceDriverUser, not SceRtc.
but when celeste tried to fix this on db.yml they said it would "break compatibility"

honestly just use TAI_ANY_LIBRARY

**Comment by SilicaAndPina** [comment]
i have allready managed to spoof sceRtcGetCurrentNetworkTick, i just dont know what module is trying to use the func

**Comment by 4rchitect** [comment]
how many modules are out there? I can brute force it.

**Comment by SilicaAndPina** [comment]
you could try just doing it for case: default.

**Comment by thintei** [comment]
How is progress on the timestamp/netclock editing?

**Comment by SilicaAndPina** [comment]
@thintei very good! In fact activate.vpk changes "secureTick" which seems to control trophy's as well as activation
